esx_service/vmdk_ops.py

```diff
@@ -489,18 +512,15 @@ def vol_info(vol_meta, vol_size_info, datastore):
     return vinfo
 
 
-# Return error, or None for OK
-def removeVMDK(vmdk_path, vol_name=None, vm_name=None, tenant_uuid=None, datastore=None):
-    logging.info("*** removeVMDK: %s", vmdk_path)
-
-    # Check the current volume status
-    kv_status_attached, kv_uuid, attach_mode = getStatusAttached(vmdk_path)
-    if kv_status_attached:
-        if handle_stale_attach(vmdk_path, kv_uuid):
-            logging.info("*** removeVMDK: %s is in use, VM uuid = %s", vmdk_path, kv_uuid)
-            return err("Failed to remove volume {0}, in use by VM uuid = {1}.".format(
-                vmdk_path, kv_uuid))
-
+def cleanVMDK(vmdk_path, vol_name=None):
+    """
+    Delete the vmdk file. Retry if the attempt fails
+    Invoked as a part of removeVMDK procedure and
+    cases requiring deletion of vmdk file only (when meta file
+    hasn't been generated)
+    eg: Unsuccesful attempt to apply vsan policy and when failed
+    to create metadata for vmdk_path
+    """
```

Comment:
Actually, removal of the volume will automatically result in the meta-data file also getting removed.. Why is this function needed?